p4-16/psa/psa.p4

```diff
@@ -568,12 +579,13 @@ control ComputeChecksum<H, M>(inout H hdr, inout M user_meta);
 
 control Deparser<H>(packet_out buffer, in H hdr);
 
-package PSA_Switch<H, M>(Parser<H, M> p,
-                         VerifyChecksum<H, M> vr,
+package PSA_Switch<H, M>(IngressParser<H, M> ip,
                          Ingress<H, M> ig,
+                         Deparser<H> igdep,
+                         EgressParser<H, M> ep,
                          Egress<H, M> eg,
                          ComputeChecksum<H, M> ck,
-                         Deparser<H> dep);
+                         Deparser<H> egdep);
```

Comment:
I would think that if you have a deparser, checksum (re)calculation should be nearby. 

Having it in a separate control is easiest.
Having it in a deparser is generally better than in ingress/egress control because we need to ensure that the fields do not change after we calculate the checksum. Imposing this restriction inside a general ingress/egress control would be difficult, but might be OK in the deparser (basically, we'd allow one extra extern there in addition to packet_out and other _outs).